src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -229,6 +230,12 @@ class ContentOpenWith extends PureComponent<Props, State> {
      */
     onIntegrationClick = (appIntegrationId: string): void => {
         const { fileId }: Props = this.props;
+        // window.open() is immediately invoked to avoid popup-blockers
+        // The name is included to be the target of a form if the integration is a POST integration
+        this.windowRef = window.open(
```

Comment:
Consider persisting the window name as an instance property and nulling out the window object in the POST case. It'll make it clear that the window object itself is still only used in the GET case, whereas the name is used for multiple things.

Reply:
I'm not sure I see the value in that. This would have to be done in the switch statement of the successhandler where we know the integration type. We're still using the window object in the POST case, but we are targeting it via its name. 

It also may be useful later if we try to minimize our creation of new windows.

Reply:
Fair enough.